### PR TITLE
[HWORKS-635] Add pytest for hsml pojo classes

### PR DESCRIPTION
**Pull request by javierdlrm**
This PR includes ~340 pytests for the POJO classes and constants (located in the hsml directory). Pytests for anything under `engine `or `core` directories are excluded in this PR.

### CODE REVIEW

**Review by vatj — APPROVED** [review]
Need to sync up the changes with the linter/formatter/pyproject PR

File: python/hsml/constants.py [hunk 1]
```diff
@@ -38,11 +38,12 @@ class ARTIFACT_VERSION:
 
 
 class RESOURCES:
+    MIN_NUM_INSTANCES = 1  # disable scale-to-zero by default
```
Comment (robzor92) [inline python/hsml/constants.py:41]: For app we do want the scale to zero enabled though if I remember correctly, is that being handled somewhere?

File: python/hsml/resources.py [hunk 2]
```diff
@@ -181,59 +187,76 @@ def _get_default_resource_limits(self):
...
             )
-        if self._limits.gpus > -1 and (
-            self._requests.gpus < 0 or self._requests.gpus > self._limits.gpus
-        ):
+        if requests.gpus < 0:
+            raise ValueError(
+                "Requested number of gpus must be greater than or equal to 0 gpus."
+            )
+        if limits.gpus > -1 and requests.gpus > limits.gpus:
             raise ValueError(
```
Comment (robzor92) [inline python/hsml/resources.py:257]: We saw before that requests.gpus and limits.gpus must always be equal. Are we adding that check in a separate PR?

File: python/tests/test_model_schema.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
+#
+#   Copyright 2022 Hopsworks AB
```
Comment (robzor92) [inline python/tests/test_model_schema.py:2]: Wrong year

**Review by robzor92 — CHANGES_REQUESTED** [review]

File: python/hsml/constants.py (hunk 1 above)
Comment (javierdlrm) [reply python/hsml/constants.py:41]: Yes, we check if scale-to-zero is required when getting the default resources [here](https://github.com/javierdlrm/machine-learning-api/blob/6e6f058875267d4e8d721c64aab3caead3bbc95b/python/hsml/predictor.py#L201) and [here](https://github.com/javierdlrm/machine-learning-api/blob/6e6f058875267d4e8d721c64aab3caead3bbc95b/python/hsml/transformer.py#L62).

While supporting non-KServe deployments, we cannot default to scale-to-zero (e.g., num_instances = 0).

File: python/hsml/resources.py (hunk 2 above)
Comment (javierdlrm) [reply python/hsml/resources.py:257]: Yes, I'm working on a different PR for that.

**Review by robzor92 — APPROVED** [review]

**Pull request merged**